niworkflows/viz/utils.py

```diff
@@ -222,19 +223,27 @@ def plot_segs(
     compress="auto",
     **plot_params
 ):
-    """ plot segmentation as contours over the image (e.g. anatomical).
+    """
+    Generate a static mosaic with ROIs represented by their delimiting contour.
+
+    Plot segmentation as contours over the image (e.g. anatomical).
     seg_niis should be a list of files. mask_nii helps determine the cut
     coordinates. plot_params will be passed on to nilearn plot_* functions. If
     seg_niis is a list of size one, it behaves as if it was plotting the mask.
     """
     plot_params = {} if plot_params is None else plot_params
 
-    image_nii = _3d_in_file(image_nii)
+    image_nii = as_canonical(_3d_in_file(image_nii))
+    seg_niis = [as_canonical(_3d_in_file(f)) for f in seg_niis]
```

Comment:
These are guaranteed to have the same affines? If not, the implicit rotation by removing the rotational components of the affine my produce a misalignment.

Might be worth having an affine check before calling `as_canonical`.

Reply:
In principle, right now it should allow you to stick in a NIfTI clipping some area (e.g., a face mask given the circumstances) with different voxel size and FoV. Not sure we are ever using this.

What if we run `as_canonical` only once and apply the same rotation matrix to all other inputs?